x-pack/plugins/ml/public/ui_actions/open_vis_in_ml_action.tsx

```diff
@@ -60,8 +73,11 @@ export function createLensVisToADJobAction(
       }
 
       try {
-        const { chartInfo } = await getJobsItemsFromEmbeddable(context.embeddable, lens);
-        return isCompatibleVisualizationType(chartInfo);
+        if (embeddableType === 'lens' && lens) {
+          const { chartInfo } = await getJobsItemsFromEmbeddable(context.embeddable, lens);
+          return isCompatibleVisualizationType(chartInfo!);
```

Comment:
I think maps should have a version of this function to ensure that the visualisation has at least one compatible layer.

Reply:
🤔 So after confirming with Nathan that any map embeddable has to have a geo field (which is all we need to create a job) I don't think we need this - we're already using this for the case where there is no index pattern.

Reply:
I think it would still be worth adding a very basic check.
If there are no layers the flyout still opens and is empty, which looks a bit odd.

<img width="1293" alt="image" src="https://user-images.githubusercontent.com/22172091/212273281-639cc4bc-f2f3-4fab-9a42-e7d0a0ac6a68.png">